### PR TITLE
Hex.Build refactoring

### PR DESCRIPTION
**Pull request by lexmag**

### CODE REVIEW

File: lib/mix/hex/build.ex [hunk 1]
```diff
@@ -151,72 +151,59 @@ defmodule Mix.Hex.Build do
...
+        []
+      missing ->
+        ["Missing metadata fields: #{Enum.join(missing, ", ")}"]
+    end
   end
 
-  defp error_long_description(meta) do
-    description = meta[:description] || ""
+  defp check_description_length(metadata) do
+    descr = metadata[:description] || ""
```
Comment (ericmj) [inline lib/mix/hex/build.ex]: Any specific reason to change these variable names?

**Comment by ericmj** [comment]
Looks great, just one minor comment.

File: lib/mix/hex/build.ex (hunk 1 above)
Comment (lexmag) [inline lib/mix/hex/build.ex]: This change is purely opinionated, I think `metadata` reads better than just `meta`.
Will change it back if you like. :bowtie:

Comment (whatyouhide) [inline lib/mix/hex/build.ex]: Personally I'm ok with `meta` or `metadata`, but I think `description` is better than `descr` :)

**Pull request merged**